Dockerfile.ci

```diff
@@ -1158,8 +1141,7 @@ COPY --from=scripts install_pip_version.sh install_airflow_dependencies_from_bra
 # the cache is only used when "upgrade to newer dependencies" is not set to automatically
 # account for removed dependencies (we do not install them in the first place)
 RUN bash /scripts/docker/install_pip_version.sh; \
-    if [[ ${AIRFLOW_PRE_CACHED_PIP_PACKAGES} == "true" && \
-          ${UPGRADE_TO_NEWER_DEPENDENCIES} == "false" ]]; then \
+    if [[ ${AIRFLOW_PRE_CACHED_PIP_PACKAGES} == "true" ]]; then \
```

Comment:
We alwaysh butild the CACHE layer from branch tip - this will not be invalidated when UPGRADE_TO_NEWER_DEPENDENCIES is set.